test/fuzz/transformation_swap_two_functions_test.cpp

```diff
@@ -0,0 +1,250 @@
+               OpReturn
+               OpFunctionEnd
+         %10 = OpFunction %6 None %8
+          %9 = OpFunctionParameter %7
+         %11 = OpLabel
+         %18 = OpLoad %6 %9
+         %20 = OpFMul %6 %18 %19
+               OpReturnValue %20
+               OpFunctionEnd
+         %13 = OpFunction %6 None %8
+         %12 = OpFunctionParameter %7
+         %14 = OpLabel
+         %23 = OpVariable %7 Function
+         %24 = OpLoad %6 %12
+               OpStore %23 %24
+         %25 = OpFunctionCall %6 %10 %23
+         %26 = OpFMul %6 %25 %19
+               OpReturnValue %26
+               OpFunctionEnd
+         %16 = OpFunction %6 None %8
+         %15 = OpFunctionParameter %7
+         %17 = OpLabel
+         %29 = OpVariable %7 Function
+         %32 = OpVariable %7 Function
+         %30 = OpLoad %6 %15
+               OpStore %29 %30
+         %31 = OpFunctionCall %6 %10 %29
+         %33 = OpLoad %6 %15
+               OpStore %32 %33
+         %34 = OpFunctionCall %6 %13 %32
+         %35 = OpFMul %6 %31 %34
+               OpReturnValue %35
+               OpFunctionEnd
+  )";
+  const auto env = SPV_ENV_UNIVERSAL_1_3; 
+  const auto consumer = nullptr;
+  const auto context = BuildModule(env, consumer, shader, kFuzzAssembleOption);
+  spvtools::ValidatorOptions validator_options;
+
+  // Check context validity.
+  ASSERT_TRUE(fuzzerutil::IsValidAndWellFormed(context.get(), validator_options,
+                                               kConsoleMessageConsumer));
+
+  TransformationContext transformation_context(
+      MakeUnique<FactManager>(context.get()), validator_options);
+
+  // Function should not swap with itself
+
+  
+  // Permutation has invalid values 2
+  ASSERT_DEATH( TransformationSwapTwoFunctions(4,4)
+                   .IsApplicable(context.get(), transformation_context),
+               "Two functions cannot be the same.");
+
+//   // Function with id 29 does not exist. 
+  ASSERT_DEATH( TransformationSwapTwoFunctions(10,29)
+                   .IsApplicable(context.get(), transformation_context),
+               "Function 2 is not in range.");
+
+//   // Function with id 30 does not exist. 
+  ASSERT_DEATH( TransformationSwapTwoFunctions(30,13)
+                   .IsApplicable(context.get(), transformation_context),
+               "Function 1 is not in range.");
+
+//   // Both function 5 and 6 do not exist. 
+  ASSERT_DEATH( TransformationSwapTwoFunctions(5,6)
+                   .IsApplicable(context.get(), transformation_context),
+               "Both functions are not in range.");
+
+//   // Function with result_id 10 and 13 should swap successfully.
+  auto swap_test5 = TransformationSwapTwoFunctions(10, 13);
+  ASSERT_TRUE(
+      swap_test5.IsApplicable(context.get(), transformation_context));
+  
+  std::string after_transformation = R"(
+               OpCapability Shader
+          %1 = OpExtInstImport "GLSL.std.450"
+               OpMemoryModel Logical GLSL450
```

Comment:
The entire shader text is identical to the above shader text. Thus, the test is wrong. You need to swap function 10 and 13 in the shader text; that is what we expect to see.

Also, you did not `Apply` the `swap_test5` transformation; you can do this by calling: `ApplyAndCheckFreshIds(swap_test5, context.get(), &transformation_context);` 

However, even after doing this locally, the module does not seem to change. Thus, the test will fail. But it would still be good to fix the test first (so that it fails).

Reply:
This is strange because it seems ``ASSERT_TRUE`` applied the change, since the change matched my new shader script with is assembled after I switched the functions.

Reply:
Fixed unit test and should be working now.